scanner/cmd/scanner/main.go

```diff
@@ -160,16 +143,16 @@ func createBackends(ctx context.Context) (*Backends, error) {
 }
 
 func (b *Backends) Close(ctx context.Context) {
-	if b.Matcher != nil {
-		err := b.Matcher.Close(ctx)
-		if err != nil {
-			zlog.Error(ctx).Err(err).Msg("closing matcher")
-		}
-	}
```

Comment:
Matcher was initialized last, so it is closed first. Why are you changing the order?

Reply:
asked about it in the previous PR: https://github.com/stackrox/stackrox/pull/6663#discussion_r1266043358

Just wanted to understand why it was in that order in the first place. Makes sense now. I will revert this